src/Resolution.hs

```diff
@@ -87,14 +87,58 @@ untabifyStr size =
 untabify :: Int -> Conflict -> Conflict
 untabify = Conflict.setBodyStrings . untabifyStr
 
+data LineEnding = LF | CRLF | Mixed
+    deriving (Eq, Ord)
+
+lineEnding :: String -> LineEnding
+lineEnding x@(_:_) | last x == '\r' = CRLF
+lineEnding _ = LF
+
+lineEndings :: [String] -> LineEnding
+lineEndings [] = Mixed
+lineEndings xs =
+    foldl1 f (map lineEnding xs)
+    where
+        f Mixed _ = Mixed
+        f x c | x == c = x
+            | otherwise = Mixed
+
+allSame :: Eq a => [a] -> Bool
+allSame (x:y:rest) = x == y && allSame (y:rest)
+allSame _ = True
+
+chooseEnding :: [LineEnding] -> LineEnding
```

Comment:
Use a traversable of 3? Perhaps even one representing base,a,b that has this choice function for it? We need it in more places

Reply:
Done